reference/supervisor.md

```diff
@@ -399,9 +401,9 @@ namespace webots {
       HINGE_2_JOINT_PARAMETERS, IMMERSION_PROPERTIES, JOINT_PARAMETERS, LENS,
       LENS_FLARE, PHYSICS, RECOGNITION, SLIDER_JOINT, SOLID, SOLID_REFERENCE,
       SLOT, TRACK, TRACK_WHEEL, WORLD_INFO, ZOOM
-    };
+    } Type;
```

Comment:
I think this new way of documenting the node list introduced in #861 is very missleading for the user as it doesn't reflect the actual order in the header, and order in an enumeration is **really** important...

Reply:
Why? Users don't need to know the value of the constant and don't need to use it. They just need to know the constant name.

Reply:
Ok, I understand what you mean.
I think that it is also very important to be able to find easily a node in a list, otherwise it is useless.
But probably we should not document it as an enumeration.
Or we should change the order in nodes.h, because the current one is just a big mess.

Reply:
Because for example a user may want to use such kind of code to detect if the node is a device (there are of course other ways to know if it is a device, but this is just an example):
```
if (node->getType() >= Node::ACCELEROMETER && node->getType() <= Node::TOUCH_SENSOR)
  std::cout << This is a device ... or not maybe not. << std::endl;
```

Reply:
> Or we should change the order in nodes.h, because the current one is just a big mess.

Yes I think that would be the best solution, because currently the order in nodes.h is completely right (it even doesn't respect our grouping rule).

Reply:
Reported in omichel/webots#7420